tools/pipelines/build-docs-website-deploy.yml

```diff
@@ -0,0 +1,81 @@
+# Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+# Licensed under the MIT License.
+
+# build-docs-website-deploy pipeline
+
+name: $(Build.BuildId)
+
+schedules:
+  - cron: "0 4 * * *"
+    displayName: Daily 8pm (PST) website deploy
+    branches:
+      include:
+        - main
+    always: true
+
+trigger: none
+pr: none
+
+parameters:
+- name: deployOverride
+  displayName: Deployment Override (default = based on branch)
+  type: string
+  default: default
+  values:
+    - default
+    - skip
+    - force
+- name: deployEnvironment
+  displayName: Static web app environment to deploy to
+  type: string
+  default: new
+  values:
+    - new
+    - old
+
+variables:
+- name: isMain
+  value: ${{ eq(variables['Build.SourceBranchName'], 'main') }}
+- name: shouldDeploy
+  value: ${{ or(
+    eq(parameters.deployOverride, 'force'),
+    and(eq(variables.isMain, true), eq(parameters.deployOverride, 'default'))
+    )}}
+- name: deploymentToken
+  ${{ if eq( parameters['deployEnvironment'], 'new' ) }}:
+    value: "$(FLUID_WEBSITE_TORUS_API_TOKEN)"
+  ${{ if eq( parameters['deployEnvironment'], 'old') }}:
+    value: "$(AZURE_STATIC_WEB_APPS_API_TOKEN)"
+
+stages:
+- stage: deploy
+  displayName: 'Deploy website'
+  pool: Small
+  jobs:
+    - job: deploy_site
+      displayName: 'Deploy website'
+      steps:
+        - checkout: self
+          submodules: false
+          clean: true
```

Comment:
It seems this is necessary in the deployment pipeline because we need to include `docs/api` from the repo, which is not published as an artifact. Should it be, though? Single artifact that includes docs/api and docs/public, so they always go together? It'd have the advantage that this pipeline wouldn't need to check out the repo, just download the artifact generated by build-docs, and that might even be a fix? I could imagine that a docs artifact might need to "match the version" of docs/api when it was published, but if we always grab docs/api from the tip of main, it could have unexpected changes...? Or if it is by design that those files always come from the tip of main, then ignore all this 😄 .

Reply:
I ended up grouping docs/api, docs/public, and docs/data into one artifact and published it. Since `index.js` uses relative references to access `versions.json` this should give it access to the versions data.

Reply:
I'm not sure it makes a difference for a couple of reasons:

- In the deploy pipeline, I'd expect we want to download the new `fulidframework-docs-temp` artifact, no?
- Even if we did, if I understand it correctly the deployment task still only deploys docs/public and docs/api, and I'm not sure we can add anything else to it. It probably expects that those two folders contain everything needed for the site and the API to work:

```yaml
  app_location: 'docs-temp/public'
  api_location: 'docs-temp/api'
```

Do we know what other uses `versions.json` has? If it has other purposes, copying it to docs/api before publishing that artifact makes sense to me, so that the deploy pipeline can just take docs/public and docs/api from the artifact and deploy them blindly (no extra pre/post-processing steps). It leaves me slightly concerned that we have to copy that in CI, because then a local build wouldn't work the same... but I suppose this is not testable locally anyway since it depends on Azure Functions.

Reply:
I ended up adding `docs-temp/api` as a part of the artifact since we aren't importing the repo anymore. I believe that originally, `docs/api` was referring to the `docs/api` folder in the repo and the `fluideframework-docs` artifact was being downloaded directly to the `docs/public` folder. With the new artifact, it should have `versions.json`, `fluidframework-docs`, and `docs/api` code all in one artifact.

Reply:
My point is two-fold:
- That in this new pipeline we are not downloading the artifact that has docs-temp/*, we're still downloading the previous one which only has docs/public.
- That it doesn't matter that the new artifacts has docs-temp/public, docs-temp/api, and docs-temp/data, because the `AzureStaticWebApp@0` task still only lets us specify an app_location (docs-temp/public) and an api_location (docs-temp/api), which means it won't upload docs-temp/data so we end up with the same problem we have today. Instead of just including the /data folder in the artifact, what we need to do is make sure the versions.json file gets uploaded to the Azure Static Web App, and I think the only way to do that is for the file to live inside the api folder that contains the azure function. The way I see it, that folder must be _entirely_ self-contained and work without references to anything outside it.